docs/multiarch.rst

```diff
@@ -0,0 +1,1105 @@
+
+Technically, only compute nodes in the required architectures are
+needed to perform the "docker build" step. These can be arranged
+either in a single mixed-architecture cluster, or with multiple
+single-architecture clusters for each architecture, or as a mix of the
+two.
+
+Client configuration on the Koji builder will decide which cluster
+and/or `node selector`_ is required to perform image builds for each
+required platform.
+
+.. _`node selector`: https://docs.openshift.org/latest/admin_guide/managing_projects.html#developer-specified-node-selectors
+
+The orchestrator build will make use of `Server-side configuration for
+atomic-reactor`_ to discover which worker clusters to direct builds
+to.
+
+If integration with Pulp is used, there will be separate Pulp
+repositories for each image and platform, following a naming scheme
+yet to be decided. For x86_64 images the repository names will remain
+unchanged.
+
+Orchestration required for multi-platform builds
+------------------------------------------------
+
+When a new multi-platform build is required, one designated cluster
+will always be asked to perform it. This cluster will perform this
+using atomic-reactor running in "orchestration" mode, i.e. with a
+different set of steps running as plugins:
+
+- For base images, create the base image filesystems for all required
+  architectures using Koji
+- Orchestrate worker builds on other clusters (or this cluster) as
+  needed; if any worker build fails the orchestrator plugin must also
+  fail (but other worker builds should be left to continue); if the
+  whole build is cancelled, worker builds will be cancelled
+- Collect package content from all builds and compare to ensure common
+  components are exactly matching (for RPMs: version-release and
+  keys used for signatures)
+- Publish Pulp repositories that had content added (if Pulp
+  integration is enabled)
+- Create a Koji Build (if Koji integration is enabled). Note: no need
+  to upload image tar archives as worker builds have done this
+- On failure, remove added Pulp content and revert Pulp docker tags
+  (if Pulp integration is enabled)
+- Update this OpenShift Build with annotations about output,
+  performance, errors, worker builds used, etc
+
+The orchestration step will create OpenShift builds for performing the
+worker builds. Inside each of these platform-specific OpenShift
+builds, atomic-reactor will execute these steps as plugins:
+
+- Make alterations to the Dockerfile, including transforming the
+  "FROM" image name and name label into a platform-specific name, as
+  well as setting the ``architecture`` label appropriately
+- "docker build"
+- Squash the image
+- Compress the docker image tar archive
+- Upload this compressed tar archive to Pulp (if Pulp integration and
+  v1 support are both enabled)
+- Tag and push the image to the docker registry, using a unique tag
+  including the timestamp, ``latest``, $version, $version-$release,
```

Comment:
Like `$version`? These are actually labels not environment variables, and I should probably make that clearer.